src/components/SendFlow/MultisigAccount/SignTransactionFormPage.test.tsx

```diff
@@ -40,45 +40,54 @@ const props: SignPageProps<FormValues> = {
 };
 
 describe("<SignPage />", () => {
-  it("has a title", () => {
+  it("has a title", async () => {
     render(fixture(props));
 
-    expect(screen.getByTestId("sign-page-header")).toHaveTextContent("Review & Submit");
+    await waitFor(() =>
+      expect(screen.getByTestId("sign-page-header")).toHaveTextContent("Review & Submit")
+    );
   });
 
-  it("has a subtitle", () => {
+  it("has a subtitle", async () => {
     render(fixture(props));
 
-    expect(
-      screen.getByText("Please review the details and then continue to submit contract.")
-    ).toBeInTheDocument();
+    await screen.findByText("Please review the details and then continue to submit contract.");
   });
 
-  it("displays the contract name", () => {
+  it("displays the contract name", async () => {
     render(fixture(props));
 
-    expect(screen.getByTestId("contract-name")).toHaveTextContent("Contract name");
+    await waitFor(() =>
+      expect(screen.getByTestId("contract-name")).toHaveTextContent("Contract name")
+    );
   });
 
-  it("displays the approvers", () => {
+  it("displays the approvers", async () => {
     render(fixture(props));
 
-    props.data.signers.forEach(signer => {
-      expect(screen.getByTestId(`approver-${signer.val}`)).toBeInTheDocument();
+    await waitFor(() => {
+      props.data.signers.forEach(signer => {
+        expect(screen.getByTestId(`approver-${signer.val}`)).toBeInTheDocument();
+      });
     });
   });
 
-  it("displays the threshold", () => {
+  it("displays the threshold", async () => {
     render(fixture(props));
 
-    expect(screen.getByTestId("threshold")).toHaveTextContent("No. of approvals:");
-    expect(screen.getByTestId("threshold")).toHaveTextContent("1 out of 2");
+    await waitFor(() => {
+      expect(screen.getByTestId("threshold")).toHaveTextContent("No. of approvals:");
+    });
+
+    await waitFor(() => {
```

Comment:
this `awaitFor` is not needed as the previous one has already processed "all the actions"